src/Symfony/Component/Cache/CacheItem.php

```diff
@@ -100,6 +100,44 @@ public function expiresAfter($time)
     }
 
     /**
+     * {@inheritdoc}
+     */
+    public function tag($tag)
+    {
+        $tag = self::normalizeTag($tag);
+        $this->tags[$tag] = $tag;
+
+        return $this;
+    }
+
+    /**
+     * Validates a cache tag.
+     *
+     * @param string $tag The tag to validate.
+     *
+     * @throws InvalidArgumentException When $tag is not valid.
+     */
+    public static function normalizeTag($tag)
+    {
+        if (!is_string($tag)) {
+            throw new InvalidArgumentException(sprintf('Cache tag must be string, "%s" given', is_object($tag) ? get_class($tag) : gettype($tag)));
+        }
+        $tag = trim($tag, '/');
+
+        if (!isset($tag[0])) {
+            throw new InvalidArgumentException('Cache tag length must be greater than zero');
+        }
+        if (isset($tag[strcspn($tag, '{}()*\@:')])) {
```

Comment:
here, `/` is not a reserved char anymore (of course since it's the hierachy separator), but `*` is, because I might need it in a later PR for adding "any-of" invalidation style (at least I missed the feat. a few times in patchwork, even though we can work around not having it). More to come in a few weeks :)